packages/slice-machine/src/modules/simulator/index.ts

```diff
@@ -308,14 +215,21 @@ function* trackOpenSetupDrawerSaga() {
 
 // Saga watchers
 function* watchCheckSetup() {
-  yield takeLatest(
+  yield takeLeading(
```

Comment:
❓ out of interest, why use `takeLeading` here instead of `takeLatest`

Reply:
`takeLatest` takes (well) the latest fork and cancels any previous call. `takeLeading` prevents any subsequent fork while the saga is playing!

Reply:
interesting